app/models/topic_tracking_state.rb

```diff
@@ -104,6 +105,25 @@ def self.publish_muted(topic)
     MessageBus.publish("/latest", message.as_json, user_ids: user_ids)
   end
 
+  def self.publish_unmuted(topic)
+    return if !SiteSetting.mute_all_categories_by_default
+    user_ids = User
+      .joins("LEFT JOIN category_users ON category_users.user_id = users.id AND category_users.category_id = #{topic.category_id}")
```

Comment:
To avoid SQL injection it's safer here to either add a `to_i` at the end of the interpolated string, or move it to be a `where` condition so you may use named parameters.

Reply:
An alternative here that is a bit more tidy is to lean on `DB.sql_fragment`

```
[1] pry(main)> DB.sql_fragment('select ?', 1)
=> "select 1"
[2] pry(main)> DB.sql_fragment('select :a', a: 1)
=> "select 1"

```

Related we use `ActiveRecord::Base.sanitize_sql_array` in a few spots ... it is a handful to type so we could just port all these over as well to this centralized helper that is a bit cleaner and faster than the Rails implementation.

It feels more tidy than the `.to_i`

Reply:
I will use `sql_fragment` just here and will create another PR to replace all `sanitize_sql_array` with `sql_fragment`